planning/costmap_generator/test/test_points_to_costmap.cpp

```diff
@@ -0,0 +1,226 @@
+  in_sensor_points.points[0].y = 1;
+  in_sensor_points.points[0].z = 1;  // out of [maximum_height_thres,minimum_lidar_height_thres]
+
+  grid_map::GridMap gridmap = constrcut_gridmap();
+
+  PointsToCostmap point2costmap;
+  double maximum_height_thres = 0.99;
+  double minimum_lidar_height_thres = 0;
+  double grid_min_value = 0;
+  double grid_max_value = 1;
+
+  std::string gridmap_layer_name = "points";
+  grid_map::Matrix costmap_data = point2costmap.makeCostmapFromPoints(
+    maximum_height_thres, minimum_lidar_height_thres, grid_min_value, grid_max_value, gridmap,
+    gridmap_layer_name, in_sensor_points);
+
+  int nonempty_gridcell_num = 0;
+  for (int i = 0; i < costmap_data.rows(); i++) {
+    for (int j = 0; j < costmap_data.cols(); j++) {
+      if (costmap_data(i, j) == grid_max_value) {
+        nonempty_gridcell_num += 1;
+      }
+    }
+  }
+
+  EXPECT_EQ(nonempty_gridcell_num, 0);
+}
+
+TEST_F(PointsToCostmapTest, TestmakeCostmapFromPoints_invalidPoints_lessThanMinimumHeightThres)
+{
+  // construct pointcloud in map frame
+  pointcloud in_sensor_points;
+  in_sensor_points.width = 1;
+  in_sensor_points.height = 1;
+  in_sensor_points.is_dense = false;
+  in_sensor_points.resize(in_sensor_points.width * in_sensor_points.height);
+
+  in_sensor_points.points[0].x = 0.7;
+  in_sensor_points.points[0].y = 1;
+  in_sensor_points.points[0].z = -0.1;  // out of [maximum_height_thres,minimum_lidar_height_thres]
+
+  grid_map::GridMap gridmap = constrcut_gridmap();
+
+  PointsToCostmap point2costmap;
+  double maximum_height_thres = 0.99;
+  double minimum_lidar_height_thres = 0;
+  double grid_min_value = 0;
+  double grid_max_value = 1;
+
+  std::string gridmap_layer_name = "points";
+  grid_map::Matrix costmap_data = point2costmap.makeCostmapFromPoints(
+    maximum_height_thres, minimum_lidar_height_thres, grid_min_value, grid_max_value, gridmap,
+    gridmap_layer_name, in_sensor_points);
+
+  int nonempty_gridcell_num = 0;
+  for (int i = 0; i < costmap_data.rows(); i++) {
+    for (int j = 0; j < costmap_data.cols(); j++) {
+      if (costmap_data(i, j) == grid_max_value) {
+        nonempty_gridcell_num += 1;
+      }
+    }
+  }
+
+  EXPECT_EQ(nonempty_gridcell_num, 0);
+}
+
+TEST_F(PointsToCostmapTest, TestmakeCostmapFromPoints_invalidPoints_outOfGrid)
+{
+  // construct pointcloud in map frame
+  pointcloud in_sensor_points;
+  in_sensor_points.width = 1;
+  in_sensor_points.height = 1;
+  in_sensor_points.is_dense = false;
+  in_sensor_points.resize(in_sensor_points.width * in_sensor_points.height);
+
+  in_sensor_points.points[0].x = 1 + grid_length_x_ / 2.0;
+  in_sensor_points.points[0].y = 1 + grid_length_y_ / 2.0;
+  in_sensor_points.points[0].z = 0.5;
+
+  grid_map::GridMap gridmap = constrcut_gridmap();
+
+  PointsToCostmap point2costmap;
+  double maximum_height_thres = 0.99;
+  double minimum_lidar_height_thres = 0;
+  double grid_min_value = 0;
+  double grid_max_value = 1;
```

Comment:
If these are constant, it's better to use `constexpr` or at least `const`.

```suggestion
  constexpr double maximum_height_thres = 0.99;
  constexpr double minimum_lidar_height_thres = 0;
  constexpr double grid_min_value = 0;
  constexpr double grid_max_value = 1;
```